### DISCUSSION TITLE
use_robust_matcher_for_relocalization_request

### DISCUSSION THREAD

**Discussion by abylikhsanov** [stella-cv/stella_vslam#198] [Q&A] [answered]
Hi,

I have noticed your recent commit where you have included a new parameter:
`use_robust_matcher_for_relocalization_request`

Is there any documentation describing what does this parameter really do?

**Comment by ymd-stella** [comment, marked as answer]
Description is https://openvslam-community.readthedocs.io/en/latest/parameters.html#tracking.

> If true, use robust_matcher for relocalization request.

Do you need more information than this?

**Reply by abylikhsanov** [reply]
Yes, I see this paramenter but does it really do? As I understand OpenVSLAM has two choices of using the localization algorithm?

**Reply by ymd-stella** [reply]
Initially, relocalization was performed with the following matching algorithm, which is fast but relatively low performance.

https://github.com/OpenVSLAM-Community/openvslam/blob/c7c8ecc0a73727fd7528be182c9d9b463157cb52/src/openvslam/match/bow_tree.cc#L16-L122

It is now possible to change it to the following matching algorithm, which is slower but has relatively high performance. I call this `robust_matcher` in description.

https://github.com/OpenVSLAM-Community/openvslam/blob/c7c8ecc0a73727fd7528be182c9d9b463157cb52/src/openvslam/match/robust.cc#L180-L213

However, the robust matcher will be used only when the following functions are called, and the fast matcher will be used as before for automatic relocalization.

https://github.com/OpenVSLAM-Community/openvslam/blob/c7c8ecc0a73727fd7528be182c9d9b463157cb52/src/openvslam/system.cc#L293-L315
